test/unit/functions/noSelfAnnotation.spec.ts

```diff
@@ -62,3 +62,30 @@ test("explicit this parameter respected over @noSelf", () => {
         export const result = foo(1);
     `.expectToMatchJsResult();
 });
+
+test("respect noSelfInFile over noImplicitSelf", () => {
+    const result = util.testModule`
+        /** @noSelfInFile **/
+
+        const funcByValue: Record<string, Function> = {
```

Comment:
Why complicate this with a record? Can't you just write `const func: Function = () => 1;`

Reply:
Im unsure if the issue really occurs then, I could check, that was the example provided by the Github Issue atleast.

Reply:
Nevermind, seems like that is possible, odd why the issue would have typed it out in that way then.